lib/buffer.js

```diff
@@ -1296,3 +1255,64 @@ Buffer.prototype.writeDoubleBE = function writeDoubleBE(val, offset, noAssert) {
     binding.writeDoubleBE(this, val, offset, true);
   return offset + 8;
 };
+
+const swap16n = binding.swap16;
+const swap32n = binding.swap32;
+const swap64n = binding.swap64;
+
+function swap(b, n, m) {
+  const i = b[n];
+  b[n] = b[m];
+  b[m] = i;
+}
+
+Buffer.prototype.swap16 = function swap16() {
+  // For Buffer.length < 128, it's generally faster to
+  // do the swap in javascript. For larger buffers,
+  // dropping down to the native code is faster.
+  const len = this.length;
+  if (len % 2 !== 0)
+    throw new RangeError('Buffer size must be a multiple of 16-bits');
+  if (len < 128) {
+    for (var i = 0; i < len; i += 2)
+      swap(this, i, i + 1);
+    return this;
+  }
+  return swap16n.apply(this);
```

Comment:
missed this on the original swap pr. no need to use `.apply()`. just pass `this` as an argument and access it via `args[0]` in c++. rest of the code does this.